CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  APIClientError.swift
+//  CreatubblesAPIClient
+//
+//  Copyright (c) 2016 Creatubbles Pte. Ltd.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import ObjectMapper
+
+//MARK: Defaults
+extension APIClientError
+{
+    public static let DefaultDomain: String = "com.creatubbles.apiclient.errordomain"
+    public static let DefaultStatus: Int    = -6000
+    public static let DefaultCode:   String = "creatubbles-apiclient-default-code"
+    public static let DefaultTitle:  String = "creatubbles-apiclient-default-title"
+    public static let DefaultSource: String = "creatubbles-apiclient-default-source"
+    public static let DefaultDetail: String = "creatubbles-apiclient-default-detail"
+}
+
+//MARK: Error codes
+extension APIClientError
+{
+    static let UnknownStatus: Int = -6001
+    static let LoginStatus: Int = -6002
+    static let UploadCancelledStatus: Int = -6003
+}
+
+public class APIClientError: ErrorType
```

Comment:
Add URL for APIClientError documentation:
https://partners.creatubbles.com/api/#errors
http://jsonapi.org/format/#error-objects